File: flutter/lib/omdb/OmdbRepository.dart
import 'dart:convert';

import 'package:http/http.dart' as http;
import 'package:mvvm_sample/omdb/models/Movie.dart';
import 'package:mvvm_sample/omdb/models/SearchResponse.dart';

class OmdbRepository {
  final apiKey = "86850a46";
  const OmdbRepository();

  Future<List<Movie>> searchMovies(String byName) async {
    final response = await http.get('http://www.omdbapi.com/?s=$byName&apikey=$apiKey');
    final jsonValue = json.decode(response.body);
    return SearchResponse.fromJson(jsonValue).movies;
  }
}

File: android/app/src/main/java/com/github/florent37/mvvmsample/omdb/OmdbRepository.kt
package com.github.florent37.mvvmsample.omdb

class OmdbRepository(val omdbApi: OMDBApi) {
    suspend fun searchMovies(byName: String): List<Movie> {
        val response = omdbApi.searchMovies(byName, OMDBApi.apiKey).await()
        return response.movies
    }
}